geowebcache/core/src/main/java/org/geowebcache/storage/BlobStore.java

```diff
@@ -136,6 +207,55 @@ public boolean deleteByGridsetId(final String layerName, final String gridSetId)
 	 */
 	public boolean layerExists(String layerName);
 
+    Map<String,Optional<Map<String, String>>> getParametersMapping(String layerName);
+    
+    /**
+     * If the given layer is cached, remove 
+     * @param layer
+     * @throws StorageException
+     */
+    public default boolean purgeOrphans(TileLayer layer) throws StorageException {
+        // TODO maybe do purging based on gridset and format
+        try{
+            final List<ParameterFilter> parameterFilters = layer.getParameterFilters();
+            
+            return getParametersMapping(layer.getName()).entrySet().stream()
+                .filter(parameterMapping -> {
+                    return parameterMapping.getValue()
+                        .map(parameters-> {
+                            // We know what the original parameters are
+                            return parameters.size() != parameterFilters.size() || // Should have the same number of parameters as the layer has filters
+                                parameterFilters.stream()
+                                    .allMatch(pfilter -> {
+                                        final String key = pfilter.getKey();
+                                        final String value = parameters.get(key);
+                                        if(Objects.isNull(value)) {
+                                            return true; // No parameter for this filter so purge
+                                        }
+                                        return !pfilter.isFilteredValue(value); // purge if it's not a filtered value
+                                    });
+                            })
+                        .orElse(true); // Don't have the original values so purge
+                    })
+                .map(Map.Entry::getKey)
+                .map(id->{
```

Comment:
The formatting seems off in many places, mostly lack of spaces were I'd exepct to find some (compared to gt/gs conventions), not sure if GWC is following the same rules though.